lib/data_services_api/service.rb

```diff
@@ -115,7 +115,11 @@ def set_connection_timeout(conn) # rubocop:disable Naming/AccessorMethodName
 
     def ok?(response, http_url)
       unless (200..207).cover?(response.status)
-        msg = "Failed to read from #{http_url}: #{response.status.inspect}"
+        response_body = JSON.parse(response.body, symbolize_names: true)
+        response_message = response_body[:message]
+        response_error = response_body[:error]
```

Comment:
Is the error component in the JSON response the same as the http status code? If not might be helpful to also capture the status code.

Reply:
@der the `msg` is what's displayed to the user and I wasn't sure if it made sense to be included as it may not make sense to them; however, the status **_is_** shown in the logging and adheres to the #104/#111 [ticket improvements](https://github.com/epimorphics/data_services_api/blob/a0c4d00bec2f43d3383418b39fe633cb0a958b2b/lib/data_services_api/service.rb#L154-L174).

Reply:
:+1: